### DISCUSSION TITLE
oramar modules separation

### DISCUSSION THREAD

**Discussion by avico78** [collerek/ormar#152] [Q&A] [answered]
On distributed env (multiple DBs ) 
it would be easier if ormar modules can be separated by some logical separation .

i.e - if we use ormar to copy/read data from multiple (DB_A ,DB_B)
it make sense to separate  the modules.py to multiple modules scripts

modules_db_b.py
```
import ormar
**from databases_config.db import BaseMetaDba**

class Test(ormar.Model):
    class Meta(**BaseMetaDba**):
        tablename: str = "table_test"

    id: int = ormar.Integer(primary_key=True)
```

modules_db_b.py
```
import ormar
from databases_config.db import **BaseMetaDbb**

class Test(ormar.Model):
    class Meta(BaseMetaDbb):
        tablename: str = "table_test"

    id: int = ormar.Integer(primary_key=True)
```

Where in databases_config.db
we define each conneciton and database separately .
databases_config.db

````
   
database_db_a = databases.Database(database_db_a _string)
metadata_db_a = sqlalchemy.MetaData()

class BaseMetaDba(ormar.ModelMeta):
    database = database_db_a 
    metadata = metadata_db_a 


   
database_db_b = databases.Database(database_db_b _string)
metadata_db_b = sqlalchemy.MetaData()

class BaseMetaDbb(ormar.ModelMeta):
    database = database_db_b
    metadata = metadata_db_b 

````

far as i tried it seems it doesn't work only if all ormar modules in same script

**Comment by collerek** [comment, marked as answer]
Can you be more specific as what is not working?

Did you create tables for both metadatas separately?

Did you connect both databases in startup/shutdow events?

I did not test multiple databases myself but it should work normal.

**Reply by avico78** [reply]
@collerek , correct no issue working with multiple databases .
